### PR TITLE
Change subscription store to support hybrid subscription resolve method

### PR DESCRIPTION
**Pull request by akalankapagoda**
Change subscription store to support hybrid subscription resolve method using direct maps for direct subscriptions and bitmaps for wildcard subscriptions.

### CODE REVIEW

**Comment by akalankapagoda** [comment]
This will provide the feasibility to handle subscriptions for each protocol individually.
Direct destinations subscriptions will be saved in a map of <destination, Set<Subscriptions>> so we can resolve subscriptions for a destination from a constant time.
Wildcard subscriptions will go to a specific subscription handler according to their protocol and can later be retrieved via ClusterSubscriptionProcessor.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/QueueManagementInformationMBean.java [hunk 1]
```diff
@@ -402,7 +402,7 @@ public long getMessageCount(String queueName,String msgPattern) {
      */
     public int getSubscriptionCount( String queueName){
         try {
-            return AndesContext.getInstance().getSubscriptionStore().numberOfSubscriptionsInCluster(queueName, false);
+            return AndesContext.getInstance().getSubscriptionStore().numberOfSubscriptionsInCluster(queueName, false, AndesSubscription.SubscriptionType.AMQP);
```
Comment (pumudu88) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/QueueManagementInformationMBean.java]: can we break this line 405 into two ? it's too long.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/SubscriptionManagementInformationMBean.java [hunk 2]
```diff
@@ -51,7 +52,7 @@ public String getObjectInstanceName() {
             List<String> allQueues = AndesContext.getInstance().getAMQPConstructStore().getQueueNames();
 
             for (String queue : allQueues) {
-                List<AndesSubscription> subscriptions = AndesContext.getInstance().getSubscriptionStore().getAllSubscribersForDestination(queue, false);
+                Set<AndesSubscription> subscriptions = AndesContext.getInstance().getSubscriptionStore().getAllSubscribersForDestination(queue, false, AndesSubscription.SubscriptionType.AMQP);
```
Comment (pumudu88) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/SubscriptionManagementInformationMBean.java]: can we break lengthy lines into 2 or more?

File: modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/SubscriptionManagementInformationMBean.java [hunk 3]
```diff
@@ -89,12 +90,12 @@ public String getObjectInstanceName() {
 
             for (String topic : allTopics) {
 
-                List<AndesSubscription> subscriptions;
+                Set<AndesSubscription> subscriptions;
                 if (!isDurableTopic) {
-                    subscriptions = AndesContext.getInstance().getSubscriptionStore().getAllSubscribersForDestination(topic, true);
+                    subscriptions = AndesContext.getInstance().getSubscriptionStore().getAllSubscribersForDestination(topic, true, AndesSubscription.SubscriptionType.AMQP);
```
Comment (pumudu88) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/SubscriptionManagementInformationMBean.java]: can we break lengthy lines into 2 or more?

File: modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/QueueManagementInformationMBean.java (hunk 1 above)
Comment (akalankapagoda) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/QueueManagementInformationMBean.java]: yes. will do.

**Pull request merged**